runtime/common/src/traits.rs

```diff
@@ -135,6 +135,14 @@ pub trait Leaser {
 
 	/// Returns the current lease period.
 	fn lease_period_index() -> Self::LeasePeriod;
+
+	/// Returns true if the parachain already has a lease in any of lease periods in the inclusive
+	/// range `[first_period, last_period]`.
```

Comment:
I think this may be slightly incorrect: technically our implementation in slots only can check the current lease period and onwards. So if any periods in `first_period..=last_period` are in the past we will not be able to look at those. Additionally we will always return `false` if the entire range is in the past (I don't think this is an issue but just wanted to point it out)